### ISSUE TITLE
Typescript: Use Generics for DndEvent

### ISSUE THREAD

**Issue by c00** [isaacHagoel/svelte-dnd-action#381]
When using typescript, the `Item` definition becomes difficult to work around in some cases.

To reproduce: Create a list with a typed list of items. Then in the `handleSort` function, try and assigning the `event.detail.items` back to the types `items`.

Typescript will complain that the type `Item` and `SomeCustomType` are not compatible. 

Example code:

```ts
<script lang="ts">
	import { dndzone } from 'svelte-dnd-action';
	import { flip } from 'svelte/animate';

  interface Dog {
    id: number;
    name: string;
    breed: string;
  }

	let items: Dog[] = [
		{ id: 1, name: 'Fido', breed: 'bulldog' },
		{ id: 2, name: 'Spot', breed: 'labrador' },
		{ id: 3, name: 'Jacky', breed: 'golden retriever' }
	];

	function handleSort(e: CustomEvent<DndEvent>) {
		items = e.detail.items;
	}
</script>

<section
	use:dndzone={{
		items,
		flipDurationMs: 100
	}}
	on:consider={handleSort}
	on:finalize={handleSort}
	class="min-h-0 transition-all w-full"
>
	{#each items as item (item.id)}
		<div animate:flip={{ duration: 100 }} class="flex">
			<span>{item.name}, {item.breed}</span>
		</div>
	{/each}
</section>
```

Resulting error:

![image](https://user-images.githubusercontent.com/6259056/174788696-02eef22c-294f-439a-ada8-22ee1b3c4b2b.png)

This is easily overcome by using generics. Something like:

```ts
	function handleSort(e: CustomEvent<DndEvent<Dog>>) {
		items = e.detail.items;
	}
```

I will add a PR for this.

**Comment by khaki32** [comment]
Yeah I ran into this issue as well. I used casting (I think?) to fix it:

```typescript
function handleSort(event: CustomEvent<DndEvent>) {
    $tasks = event.detail.items as Task[];
  }
```
But it's a lot nicer with generics.

**Comment by mellanslag-de** [comment]
(just a notice that this can probably be closed as well, in addition to #382 )

**Comment by isaacHagoel** [comment]
@mellanslag-de true. closing. thanks
